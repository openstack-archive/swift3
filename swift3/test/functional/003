#!/bin/bash

# GET Bucket

. ./common

S3USER=tester

_s3_put /bucket
_s3_put /bucket/sample.jpg -T /dev/null
_s3_put /bucket/photos/2006/January/sample.jpg -T /dev/null
_s3_put /bucket/photos/2006/February/sample2.jpg -T /dev/null
_s3_put /bucket/photos/2006/February/sample3.jpg -T /dev/null
_s3_put /bucket/pho/2006/February/sample4.jpg -T /dev/null

_s3_get /bucket?delimiter=/\&prefix=photos/2006/ -D - | _filter_curl xml
