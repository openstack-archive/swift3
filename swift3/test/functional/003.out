QA output created by 003
> s3curl --id tester -- -X PUT http://SWIFT_HOST/bucket... 200
> s3curl --id tester -- -X PUT -T /dev/null http://SWIFT_HOST/bucket/sample.jpg... 200
> s3curl --id tester -- -X PUT -T /dev/null http://SWIFT_HOST/bucket/photos/2006/January/sample.jpg... 200
> s3curl --id tester -- -X PUT -T /dev/null http://SWIFT_HOST/bucket/photos/2006/February/sample2.jpg... 200
> s3curl --id tester -- -X PUT -T /dev/null http://SWIFT_HOST/bucket/photos/2006/February/sample3.jpg... 200
> s3curl --id tester -- -X PUT -T /dev/null http://SWIFT_HOST/bucket/pho/2006/February/sample4.jpg... 200
> s3curl --id tester -- -X GET -D - http://SWIFT_HOST/bucket?delimiter=/&prefix=photos/2006/... 200
HTTP/1.1 200 OK
Content-Length: LENGTH
Content-Type: application/xml
Date: DATE
X-Trans-Id: TXID

<?xml version='1.0' encoding='UTF-8'?>
<ListBucketResult xmlns="http://s3.amazonaws.com/doc/2006-03-01/">
  <Name>bucket</Name>
  <Prefix>photos/2006/</Prefix>
  <Marker/>
  <MaxKeys>1000</MaxKeys>
  <Delimiter>/</Delimiter>
  <IsTruncated>false</IsTruncated>
  <CommonPrefixes>
    <Prefix>photos/2006/February/</Prefix>
  </CommonPrefixes>
  <CommonPrefixes>
    <Prefix>photos/2006/January/</Prefix>
  </CommonPrefixes>
</ListBucketResult>
